### DISCUSSION TITLE
No module named myhdl.conversion.iverilog

### DISCUSSION THREAD

**Discussion by Kreijstal** [myhdl/myhdl#436] [Q&A] [answered]
but 
```
kreijstal@kreijstalnuc:~/git/workspace/cosim$ ls /usr/lib/python3.12/site-packages/myhdl
_always_comb.py  conversion            _getcellvars.py   _misc.py          _Signal.py        _visitors.py
_always.py       _Cosimulation.py      _getHierarchy.py  _modbv.py         _Simulation.py    _Waiter.py
_always_seq.py   debug.py              __init__.py       _openport.py      _simulator.py
_bin.py          _delay.py             _instance.py      __pycache__       _traceSignals.py
_block.py        _enum.py              _intbv.py         _resolverefs.py   _tristate.py
_concat.py       _extractHierarchy.py  _join.py          _ShadowSignal.py  _util.py
kreijstal@kreijstalnuc:~/git/workspace/cosim$ ls /usr/lib/python3.12/site-packages/myhdl/conversion/
_analyze.py  _misc.py     _toVerilog.py      _toVHDL.py  _VHDLNameValidation.py
__init__.py  __pycache__  _toVHDLPackage.py  _verify.py
kreijstal@kreijstalnuc:~/git/workspace/cosim$
```

Why don't I have this? I installed myhdl from pip

**Comment by josyb** [comment]
_this_ being?

**Comment by josyb** [comment, marked as answer]
Excuses, I wasn't fully awake yet ...
Inferring from  your directory: xx/cosim I assume you want to run cosimulation. I just installed Python3.12 on a virgin machine and installed MyHDL with pip and get the same files as you got (evidently it can only be the same). And I noticed that the cosimulation files are not a part of this installation. For this you need to clone the  myhdl repo from GitHub. You activated issue #437 so it looks like you done that. I will continue with #437

**Reply by Kreijstal** [reply]
Pip myhdl is indeed misleading, since you don't get the vpi lol

**Reply by josyb** [reply]
lol?

**Reply by josyb** [reply]
> Pip myhdl is indeed misleading, since you don't get the vpi lol

Actually our BDFL choose not to add the cosimulation directories/files as they vary depending on the OS and the chosen simulator. As a consequence you have to build the ._vpi_ yourself. It is documented in https://docs.myhdl.org/en/stable/manual/cosimulation.html?highlight=vpi#implementation-notes and surroundings
